src/smt.rs

```diff
@@ -360,7 +367,8 @@ pub fn as_const<V: Into<SMTExpr>>(sort: SMTSort, val: V) -> SMTExpr {
     }
 }
 
-pub fn literal(lit: String, sort: SMTSort) -> SMTExpr {
+fn literal(lit: String, sort: SMTSort) -> SMTExpr {
+    assert!(lit.starts_with("#x") || lit.starts_with("(_ bv") || &lit == "true" || &lit == "false");
```

Comment:
why would it start with `(_ bv`? this interface seems odd

Reply:
`#x...` is a bitvector literal in hex notation, `(_ bv... <bits>)` is a bitvector literal in decimal notation.